core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -173,10 +178,39 @@ static __maybe_unused const char *shres2str_state(enum stm32mp_shres id)
 	return shres2str_state_tbl[id];
 }
 
+/* GPIOZ bank pin count depends on SoC variants */
+#ifdef CFG_DT
+/* A light count routine for unpaged context to not depend on DTB support */
+static int gpioz_nbpin = -1;
+
+static unsigned int get_gpioz_nbpin(void)
+{
+	if (gpioz_nbpin < 0)
+		panic();
+
+	return gpioz_nbpin;
+}
+
+static TEE_Result set_gpioz_nbpin_from_dt(void)
```

Comment:
Why isn't this called from `stm32mp1_init_shres()`?

Reply:
`stm32mp1_init_shres()` is called at the end of the driver inits, since once called, driver can no more register shared peripheral (`registering_locked=true`), hence `driver_init_late(stm32mp1_init_shres);`.
Reading gpioz pin count from DT should be done before any driver manipulate GPIOs, hence before drivers init, hence `service_init(set_gpioz_nbpin_from_dt);`.

Reply:
I see, thanks for the explanation. A comment explaining that here would be nice.